cmake/CgvFindVersion.cmake

```
@@ -69,104 +69,160 @@ if(CMAKE_SCRIPT_MODE_FILE)
   cmake_minimum_required(VERSION 3.8)
 endif()
 
+#-----------------------------------------------------------------------------#
+
+function(_cgv_store_version string suffix hash)
+  set(_CACHED_VERSION "${string}" "${suffix}" "${hash}")
+  # Note: extra 'unset' is necessary if using CMake presets with
+  # ${CGV_PROJECT}_GIT_DESCRIBE=""
+  unset(${CGV_CACHE_VAR} CACHE)
+  set(${CGV_CACHE_VAR} "${_CACHED_VERSION}" CACHE INTERNAL
+    "Version string and hash for ${CGV_PROJECT}")
```

```suggestion
  set(${CGV_CACHE_VAR} "${_CACHED_VERSION}" CACHE INTERNAL
    "Version string and hash for ${CGV_PROJECT}" FORCE)
```
More of a question than a suggestion.  Are the 2 alternatives functionally equivalent?

---

See the comment one line above: if you don't use the `unset`, for some reason the cache variable doesn't get overridden even with `INTERNAL`, (which always `FORCE`s it) if using cmake presets. It's hard to tell if that's a bug or just a weird consequence of the CMake variable scoping...

---

Fair enough (I forgot that `INTERNAL` implies `FORCE`).  (still consider adding `necessary (even with explicit INTERNAL keyword`).